### PR TITLE
Add yarn.lock v2 support

### PR DESCRIPTION
**Pull request by olexii4**
Signed-off-by: Oleksii Orel <oorel@redhat.com>

### What does this PR do?
This PR adds yarn.lock v2 support.

### What issues does this PR fix or reference?
fixes https://github.com/eclipse/che/issues/20059

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -8,11 +8,11 @@
 # Contributors:
 #   Red Hat, Inc. - initial API and implementation
 
-FROM docker.io/openjdk:15-jdk
+FROM docker.io/openjdk
```
Comment (sleshchenko) [inline Dockerfile:11]: better not to depend on the latest tag, since then it may unexpectedly got broken

File: src/bump-deps.js [hunk 2]
```diff
@@ -10,150 +10,50 @@
...
+    depsMap.set(result[1], result[2]);
   }
 }
 
 // update depsMap
-function parseDependenciesFile(fileData, dependenciesMap) {
-  let numberUnusedExcludes = 0;
-  if (dependenciesMap.size !== 0) {
+function parseDependenciesFileData(fileData, depsMap) {
+  const pattern = /^npm\/npmjs\/(-\/)?([^,]+)\/([^,]+), ([^,]+)?, approved, (\w+)$/gm;
```
Comment (sleshchenko) [inline src/bump-deps.js:29]: it's easier to understand pattern when it has examples, which the author wants to cover.

File: src/entrypoint.sh [hunk 3]
```diff
@@ -1,148 +1,20 @@
 #!/bin/bash
 
-usage() {
```
Comment (sleshchenko) [inline src/entrypoint.sh:3]: is this removal related to yarn.lock v2?

File: src/bump-deps.js (hunk 2 above)
Comment (olexii4) [reply src/bump-deps.js:29]: This PR accidentally is based not on the latest commit. I am going to rework it.

File: src/entrypoint.sh (hunk 3 above)
Comment (olexii4) [reply src/entrypoint.sh:3]: This PR accidentally is based not on the latest commit. I am going to rework it.

File: Dockerfile (hunk 1 above)
Comment (olexii4) [reply Dockerfile:11]: Ok. I will fix it.

**Review by sleshchenko — COMMENTED** [review]
Changes LGTM, I'm gonna to test it a bit later

**Review by sleshchenko — APPROVED** [review]
I can't say much about changes but I tested it on https://github.com/eclipse-che/che-dashboard/pull/288 and it seems to work just fine.

File: src/entrypoint.sh [hunk 4]
```diff
@@ -125,6 +136,7 @@ fi
 if [ -z "$CHECK" ]; then
     cp $TMP_DIR/prod.md $DEPS_DIR/prod.md
     cp $TMP_DIR/dev.md $DEPS_DIR/dev.md
+    cp $TMP_DIR/logs $DEPS_DIR/logs
```
Comment (akurinnoy) [inline src/entrypoint.sh:139]: In my understanding only `--debug` should be used to keep any temporary files and logs, while `--check` means to update nothing and `--generate` to generate/update files to have them tracked by git. If something gets wrong one can also glance at the output in the terminal :)

**Review by akurinnoy — APPROVED** [review]

File: src/entrypoint.sh (hunk 4 above)
Comment (olexii4) [reply src/entrypoint.sh:139]: I totally disagree with that. When I use this software and I need to generate a new files after adding some new dependencies. In the case when these new dependencies already have CQs - jast add changes into commit. In that case when we do not have CQs for new dependencies I need to save information about new transitive dependencies which was add and they type(prod or dev). That is why I have added a log with that information. I don't want to run this check twice. It is spend a lot of time.

@sleshchenko  WDYT?

Comment (sleshchenko) [reply src/entrypoint.sh:139]: > In that case when we do not have CQs for new dependencies I need to save information about new transitive dependencies which was add and they type(prod or dev).

isn't it already show in dependency location, I mean prod.md or dev.md? I did not get why you need logs for such case.

Comment (akurinnoy) [reply src/entrypoint.sh:139]: I'm not sure I fully got your answer. Nevertheless, this isn't crucial and these changes have already got my approval.

**Review by akurinnoy — APPROVED** [review]

**Pull request merged**